tools/boost-builder/Dockerfile

```diff
@@ -0,0 +1,9 @@
+# Does this match Travis's trusty?
+FROM ubuntu:trusty
+
+ADD build_boosts.zsh /root/
+
+RUN apt-get update
+RUN apt-get -y install build-essential python2.7-dev wget zsh
```

Comment:
It pulls in gcc-4.8. Our actual CI uses clang.

I don't think it's particularly important that these match, but this container should try to use the same compiler that Ubuntu uses to build Trusty packages. It's very likely that they are using whatever build-essential provides, but I will look for proof of that.